### ISSUE TITLE
[clinical-onset-picker] Support form validation

### ISSUE THREAD

**Issue by KunigundaLauly** [cerner/terra-clinical#806]
# Feature Request
Please add the following fields & functionality to the [onset picker component](https://engineering.cerner.com/terra-ui/components/terra-clinical-onset-picker/clinical-onset-picker/clinical-onset-picker
):
error, isInvalid, required.

[datepicker example](https://engineering.cerner.com/terra-ui/components/cerner-terra-framework-docs/date-picker/date-picker-field)

## Description

So, that it could have red border and error text, when the field is invalid.

![image](https://user-images.githubusercontent.com/47626274/165043547-825c5476-ab3c-47d7-be0c-63112d8bd424.png)

**Comment by chrismichalewicz** [comment]
UXPLATFORM-6592 created to track internally

**Comment by chrismichalewicz** [comment]
@KunigundaLauly To help with prioritization, can you provide a little more context as to why you are requesting this functionality? Is there any desired workflows that are currently blocked by not having this?

**Comment by KunigundaLauly** [comment]
@chrismichalewicz, sorry, I didn't notice your question.
We have a workflow, where the onset picker should be a required field, and on save should display an error text and the date picker (maybe its other inputs as well) with a red border.

The date picker already has this functionality:
code:
`  <Field required><DatePickerField  error={"Required"}  isInvalid={true}/></Field>
`
![image](https://user-images.githubusercontent.com/47626274/176139112-cee6be69-087e-4de1-986e-a8863582114f.png)

Currently, we are not blocked by this. We show the error text  (by wrapping it in a <Form field) and the date picker without the red border:
`  <Field required error={"Required"}  isInvalid={true}><OnsetPicker /></Field>
`
![image](https://user-images.githubusercontent.com/47626274/176140738-4412ced4-06f7-40f5-b42e-9ac4c6bd9ef0.png)

We would like to have these props in the onset picker as well.
code:
`  <Field required><OnsetPicker error={"Required"}  isInvalid={true} /></Field>
`
![image](https://user-images.githubusercontent.com/47626274/176141252-1fb447db-9db5-4ddc-8b66-42a0fc28260c.png)

Note: the required field, in the case of the date picker, is used on the container form component, which might be not necessary to be added to the onset picker.
